hunter_base/include/hunter_base/hunter_messenger.hpp

```diff
@@ -352,6 +468,30 @@ class HunterMessenger {
       phi_i *= -1.0;
     return k*phi_i;
   }
+
+  double AngelVelocity2Angel(double& linear_vel, double& angular_vel, double &radius)
+  {
+    double linear = fabs(linear_vel);
+    double angular = fabs(angular_vel);
+    if(angular == 0)
+    {
+      return 0.0;
+    }
+
+    radius = linear / angular;
+
+    int k = angular_vel / fabs(angular_vel);
+    if ((radius-l)<0 )
+    {
+      return  k*max_steer_angle;
+    }
+
+    double phi_i;
+    phi_i = atan(l/(radius-w/2));
+    if(linear_vel<0)
+      phi_i *= -1.0;
+    return k*phi_i;
+  }
```

Comment:
is this mainly to calculate the steering angle? 

may be this can improve reliability
```bash
    double linear = fabs(linear_vel);
    double angular = fabs(angular_vel);

    // If angular velocity is zero, return zero steering angle
    if (angular == 0) {
        return 0.0;
    }

    // Calculate the turning radius
    double radius = linear / angular;

    // Determine the direction of the steering angle
    int direction = angular_vel / fabs(angular_vel);

    // If the radius is less than l, return the maximum steering angle
    if ((radius - l) < 0) {
        return direction * max_steer_angle;
    }

    // Calculate the inner wheel steering angle
    double phi_i = atan(l / (radius - w / 2));
    if (linear_vel < 0) {
        phi_i *= -1.0;
    }

    // Return the calculated steering angle with the correct direction
    return direction * phi_i;
```

Reply:
@GPrathap if this is correct can you please update the PR?